packages/ckeditor5-ui/src/panel/balloon/balloonpanelview.ts

```diff
@@ -395,6 +424,11 @@ export default class BalloonPanelView extends View {
 	private _stopPinning(): void {
 		this.stopListening( global.document, 'scroll' );
 		this.stopListening( global.window, 'resize' );
+
+		if ( this._resizeObserver ) {
+			this._resizeObserver.destroy();
+			this._resizeObserver = null;
+		}
```

Comment:
Maybe we should also destroy the `ResiseObserver` instance on `destroy()`.

Reply:
Good point, I'll add it to destroy method.

Reply:
I checked it a little bit further, and it looks like `destroy` method does not unpin the tooltip properly. I adjusted destroy method and added tests.